src/ipc/apl-ipc.c

```diff
@@ -54,6 +54,8 @@
 
 extern struct ipc *_ipc;
 
+static int host_ready = 1;
```

Comment:
ok, is this being use to track whether HW is ready to send a new notification IPC to the host ? If so, we should be using HW register bits to track this and not a flag

Reply:
Please also put more comment in your logic too.

Reply:
I am sorry to explain late.
Why a static flag is used instead of register.

In IRQ function in DSP space, 

	/* reply message(done) from host */
	if (dipcie & IPC_DIPCIE_DONE) {
		/* clear DONE bit - tell host we have completed the operation */
		ipc_write(IPC_DIPCIE, ipc_read(IPC_DIPCIE) |IPC_DIPCIE_DONE);
       }

After clearing, the value of IPC_DIPCIE at bit IPC_DIPCIE_DONE is zero.  So we could check this bit to make sure host is ready. But in message sending function, the register IPC_DIPCIE is
cleared to zero to trigger a interrupt to host. So the zero value of this register doesn't mean host
is ready

void ipc_platform_send_msg(struct ipc *ipc) {
...
	/* now interrupt host to tell it we have message sent */
	ipc_write(IPC_DIPCIE, 0);
	ipc_write(IPC_DIPCI, 0x80000000 | msg->header);
...
}

We checked the register spec on APL and other platform, the register number on BYT, BDW or CNL  is different with APL. We can check register on these platforms except APL. So  we add a static flag to check.

Reply:
@lgirdwood 
Will sync with Rander to refine this patch set.
@RanderWang 
I think I can just close this PR and you continue by add the CNL version.

Reply:
no, this is for APL, I will add a PR for CNL

Reply:
@lgirdwood   yes, a register should be checked for this case. I made a mistake and choose a wrong register.

Reply:
@xiulipan ,  can you help to close it? I create another PR. Thank you!